htdocs/accountancy/class/accountancycategory.class.php

```diff
@@ -438,9 +438,14 @@ public function delete($user, $notrigger = 0)
 	public function display($id)
 	{
 		global $conf;
+
+		$pcgver = getDolGlobalInt('CHARTOFACCOUNTS');
+
 		$sql = "SELECT t.rowid, t.account_number, t.label";
 		$sql .= " FROM ".$this->db->prefix()."accounting_account as t";
-		$sql .= " WHERE t.fk_accounting_category = ".((int) $id);
+		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."accounting_system as asy ON t.fk_pcg_version = asy.pcg_version AND t.entity = ".((int) $conf->entity);
+		$sql .= " WHERE asy.rowid = ".((int) $pcgver);
```

Comment:
IF you have a WHERE, the LEFT JOIN is useless and can be replaced with INNER JOIN
The where here must be a AND in the previous line of ON ... 
Also test on t.entity must not be into the condition of the LEFT JOIN but into the WHERE.
so
```
$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."accounting_system as asy ON t.fk_pcg_version = asy.pcg_version  AND asy.rowid = ".((int) $pcgver);
$sql .= " WHERE t.entity = ".((int) $conf->entity) 
...
```